include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,432 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+                m_dArchitecture[i].TileBasedRenderer = false;
+                m_dArchitecture[i].UMA = false;
+                m_dArchitecture[i].CacheCoherentUMA = false;
+            }
+
+            m_dArchitecture1[i].NodeIndex = i;
+            if (INITIALIZE_FAILED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i])) {
+                m_dArchitecture1[i].TileBasedRenderer = false;
+                m_dArchitecture1[i].UMA = false;
+                m_dArchitecture1[i].CacheCoherentUMA = false;
+                m_dArchitecture1[i].IsolatedMMU = false;
+            }
+        }
+
+        // Initialize Feature Levels data
+        if (FAILED(m_hStatus = QueryHighestFeatureLevel())) {
+            return m_hStatus;
+        }
+        
+        return m_hStatus;
+    }
+
+    HRESULT GetStatus() const
+    {
+        return m_hStatus;
+    }
+
+    static std::variant<CD3DX12FeatureSupport, HRESULT> Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features;
+        HRESULT result = features.Init(pDevice);
+        if (FAILED(result)) {
+            return result;
+        }
+
+        return features;
+    }
+
+    // 0: D3D12_OPTIONS
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, DoublePrecisionFloatShaderOps);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, OutputMergerLogicOp);
+    FEATURE_SUPPORT_GET(D3D12_SHADER_MIN_PRECISION_SUPPORT, m_dOptions, MinPrecisionSupport);
+    FEATURE_SUPPORT_GET(D3D12_TILED_RESOURCES_TIER, m_dOptions, TiledResourcesTier);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_BINDING_TIER, m_dOptions, ResourceBindingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, PSSpecifiedStencilRefSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, TypedUAVLoadAdditionalFormats);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, ROVsSupported);
+    FEATURE_SUPPORT_GET(D3D12_CONSERVATIVE_RASTERIZATION_TIER, m_dOptions, ConservativeRasterizationTier);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions, MaxGPUVirtualAddressBitsPerResource);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, StandardSwizzle64KBSupported);
+    FEATURE_SUPPORT_GET(D3D12_CROSS_NODE_SHARING_TIER, m_dOptions, CrossNodeSharingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, CrossAdapterRowMajorTextureSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_HEAP_TIER, m_dOptions, ResourceHeapTier);
+
+    // 1: Architecture
+    // Kept for compatability with older versions
+    BOOL TileBasedRenderer(UINT NodeIndex = 0) {
+        return m_dArchitecture1[NodeIndex].TileBasedRenderer || m_dArchitecture[NodeIndex].TileBasedRenderer;
+    }
+
+    BOOL UMA(UINT NodeIndex = 0) {
+        return m_dArchitecture1[NodeIndex].UMA || m_dArchitecture[NodeIndex].UMA;
+    }
+
+    BOOL CacheCoherentUMA(UINT NodeIndex = 0) {
+        return m_dArchitecture1[NodeIndex].CacheCoherentUMA || m_dArchitecture[NodeIndex].CacheCoherentUMA;

```

Comment:
Returns true as long as one of the two Architecture struct returns true. It should work on older D3D12 without Architecture1.